cmd/openstack-cloud-controller-manager/main.go

```diff
@@ -20,240 +20,56 @@ limitations under the License.
+	"github.com/spf13/pflag"
 	"k8s.io/apiserver/pkg/server/healthz"
-	cloudprovider "k8s.io/cloud-provider"
 	"k8s.io/cloud-provider-openstack/pkg/cloudprovider/providers/openstack"
-	"k8s.io/cloud-provider-openstack/pkg/version"
-	"k8s.io/component-base/cli/flag"
 	"k8s.io/component-base/logs"
 	_ "k8s.io/component-base/metrics/prometheus/restclient" // for client metric registration
 	_ "k8s.io/component-base/metrics/prometheus/version"    // for version metric registration
 	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/cmd/cloud-controller-manager/app"
-	cloudcontrollerconfig "k8s.io/kubernetes/cmd/cloud-controller-manager/app/config"
-	"k8s.io/kubernetes/cmd/cloud-controller-manager/app/options"
-	cloudcontrollers "k8s.io/kubernetes/pkg/controller/cloud"
-	routecontroller "k8s.io/kubernetes/pkg/controller/route"
-	servicecontroller "k8s.io/kubernetes/pkg/controller/service"
 	_ "k8s.io/kubernetes/pkg/features" // add the kubernetes feature gates
-	utilflag "k8s.io/kubernetes/pkg/util/flag"
-
-	"github.com/spf13/cobra"
-	"github.com/spf13/pflag"
 )
 
 func init() {
 	mux := http.NewServeMux()
 	healthz.InstallHandler(mux)
 }
 
-var (
-	versionFlag bool
-)
-
 func main() {
 	rand.Seed(time.Now().UTC().UnixNano())
 
-	goflag.CommandLine.Parse([]string{})
-	s, err := options.NewCloudControllerManagerOptions()
-	if err != nil {
-		klog.Fatalf("unable to initialize command options: %v", err)
-	}
-
-	command := &cobra.Command{
-		Use: "openstack-cloud-controller-manager",
-		Long: `The Cloud controller manager is a daemon that embeds
-the cloud specific control loops shipped with Kubernetes.`,
-		PersistentPreRun: func(cmd *cobra.Command, args []string) {
-			// Glog requires this otherwise it complains.
-			goflag.CommandLine.Parse(nil)
-
-			// This is a temporary hack to enable proper logging until upstream dependencies
-			// are migrated to fully utilize klog instead of glog.
-			klogFlags := goflag.NewFlagSet("klog", goflag.ExitOnError)
-			klog.InitFlags(klogFlags)
-
-			// Sync the glog and klog flags.
-			cmd.Flags().VisitAll(func(f1 *pflag.Flag) {
-				f2 := klogFlags.Lookup(f1.Name)
-				if f2 != nil {
-					value := f1.Value.String()
-					f2.Value.Set(value)
-				}
-			})
-		},
-		Run: func(cmd *cobra.Command, args []string) {
-			if versionFlag {
-				version.PrintVersionAndExit()
-			}
-
-			utilflag.PrintFlags(cmd.Flags())
-
-			c, err := s.Config(KnownControllers(), ControllersDisabledByDefault.List())
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
-			}
-
-			if err := app.Run(c.Complete(), wait.NeverStop); err != nil {
-				fmt.Fprintf(os.Stderr, "%v\n", err)
-				os.Exit(1)
-			}
-		},
-	}
-
-	fs := command.Flags()
-	namedFlagSets := s.Flags(KnownControllers(), ControllersDisabledByDefault.List())
-	for _, f := range namedFlagSets.FlagSets {
-		fs.AddFlagSet(f)
-	}
-
-	fs.BoolVar(&versionFlag, "version", false, "Print version and exit")
-
-	openstack.AddExtraFlags(pflag.CommandLine)
```

Comment:
This would need to stay, it's adding this command line flag:

```
// AddExtraFlags is called by the main package to add component specific command line flags
func AddExtraFlags(fs *pflag.FlagSet) {
	fs.StringArrayVar(&userAgentData, "user-agent", nil, "Extra data to add to gophercloud user-agent. Use multiple times to add more than one component.")
}
```